File: API/Models/Function.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Models
{
    [Table("Function")]
    public partial class Function
    {
        public Function()
        {
            RoleFunctions = new HashSet<RoleFunction>();
            AccountFunctions = new HashSet<AccountFunction>();
        }

        [Key]
        public long Id { get; set; }
        [Required]
        [StringLength(250)]
        public string Area { get; set; }
        [Required]
        [StringLength(250)]
        public string Controller { get; set; }
        [Required]
        [StringLength(250)]
        public string Action { get; set; }
        [StringLength(250)]
        public string Title { get; set; }
        [StringLength(250)]
        public string Description { get; set; }
        public bool IsShow { get; set; }
        public bool? IsMenu { get; set; }
        public int? Seq { get; set; }
        public bool? IsDelete { get; set; }
        public long CreateBy { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime CreateTime { get; set; }
        public long? UpdateBy { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime? UpdateTime { get; set; }
        [InverseProperty("Function")]
        public virtual ICollection<RoleFunction> RoleFunctions { get; set; }
        [InverseProperty("Function")]
        public virtual ICollection<AccountFunction> AccountFunctions { get; set; }
    }
}

File: SPA/src/app/core/models/admins/function.ts
import { KeyValuePair } from "@utilities/key-value-utility";

export interface FunctionView {
    area: string;
    controller: string;
    title: string;
    actionDetails: ActionDetail[];
    createBy: number;
    createTime: string | Date;
    updateBy: number | null;
    updateTime: string | null | Date;
    isUpdate?: boolean;
}

export interface ActionDetail {
    id: number;
    action: string;
    description: string;
    isShow: boolean;
    isMenu: boolean | null;
    seq: number | null;
    isDelete: boolean | null;
    selection?: KeyValuePair;
    isNew?: boolean;
}

export interface Function {
    id: number;
    area: string;
    controller: string;
    action: string;
    title: string;
    description: string;
    isShow: boolean;
    isMenu: boolean | null;
    seq: number | null;
    isDelete: boolean | null;
    createBy: number;
    createTime: string | Date;
    updateBy: number | null;
    updateTime: string | null | Date;
    isUpdate?: boolean;
}

export interface FunctionParam {
    title: string;
    description: string;
    area: string;
    controller: string;
    action: string;
}

export interface FunctionSearchParam {
    area: string;
    controller: string;
    keyword: string;
}